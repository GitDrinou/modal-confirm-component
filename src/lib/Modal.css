// dialog overlay
.c-dialog {
  position: fixed;
  z-index: 100;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  display: flex;
  padding: 2.4rem;
  overflow-y: scroll;
  -webkit-overflow-scrolling: touch;
  background-color: rgba(black, .75);
  transition: .2s;         
}
  
.c-dialog__box {
  flex: 1;
  max-width: 48rem;
  margin: auto;
  padding: 2.4rem;
  background-color: white;
  font-family: 'Fjalla One', Arial, Helvetica, sans-serif;
  border-radius: 20px;
  position: relative;
}

  
.btnClose {
  position: absolute;
  right: 15px;
  top: 10px;
  font-weight: 700;
  font-family: 'Oxygen', Arial, Helvetica, sans-serif;
  border: 1px solid #000000;
  border-radius: 50%;
  background: #666666;
  font-size: 0.8rem;
  color: #FFFFFF;
}

.c-dialog[aria-hidden="true"] {
  visibility: hidden;
  opacity: 0;
}