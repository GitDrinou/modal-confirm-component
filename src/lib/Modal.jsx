import React, { useState } from 'react'
import './Modal.css'


const Modal = ({ viewModal, dialogTitle, dialogDescription, returnTo }) => {

    const isShown = viewModal

    const [isHidden, setIsHidden] = useState(isShown) 
    
    const handleClose = () => {
        if(window.location.pathname === returnTo) {
            window.location.reload(false)
        } else {
            window.location.pathname = returnTo 
        }        
        setIsHidden(true)
    }

    return(
        <div
            id="dialog"
            role="dialog"
            aria-labelledby="dialog-title"
            aria-describedby="dialog-desc"
            aria-modal="true"
            aria-hidden={isShown ? isHidden : !isHidden}
            tabIndex="-1"
            className="c-dialog">
            <div role="document" className="c-dialog__box">
                <button type="button" aria-label="Close" title="Close window" className="btnClose" onClick={handleClose}>X</button>
                <h2 id="dialog-title">{dialogTitle}</h2>
                <p id="dialog-desc">{dialogDescription}</p>
            </div>
        </div>
    )
}

export default Modal