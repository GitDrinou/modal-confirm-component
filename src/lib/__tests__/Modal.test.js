import React from "react"
import Modal from "../Modal"
import renderer from "react-test-renderer"

describe("When user click on a button and modal should be displaying", () => {
  it("should render a Modal", () => {
    const tree = renderer
      .create(<Modal viewModal="true" dialogTitle="This is my fake title" dialogDescription="This is my fake description" returnTo="/" />)
      .toJSON()
    expect(tree).toMatchSnapshot()    
  })
})

describe("When user click on a button and modal shouldn't be displaying", () => {
    it("should not render a Modal", () => {
        const tree = renderer
            .create(<Modal viewModal="false" dialogTitle="This is my fake title" dialogDescription="This is my fake description" returnTo="/" />)
            .toJSON()
          expect(tree).toMatchSnapshot()
    })
})

describe("When modal is shown, and user click on X button", () => {
    it("should hide the modal", () => {
        const onClick = jest.fn();    
        const tree = renderer
            .create(<button type="button" aria-label="Close" title="Close window" className="btnClose" onClick={onClick}>X</button>)
            .root

        const buttonInstance = tree.findByProps({
            type: 'button',
        })
        
        buttonInstance.props.onClick(onClick)
        expect(onClick).toHaveBeenCalledTimes(1)
    })
})