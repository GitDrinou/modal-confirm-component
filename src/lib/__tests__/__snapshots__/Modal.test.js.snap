// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders properly 1`] = `
<div
  aria-describedby="dialog-desc"
  aria-hidden="true"
  aria-labelledby="dialog-title"
  aria-modal="true"
  className="c-dialog"
  id="dialog"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="c-dialog__box"
    role="document"
  >
    <button
      aria-label="Close"
      className="btnClose"
      onClick={[Function]}
      title="Close window"
      type="button"
    >
      X
    </button>
    <h2
      id="dialog-title"
    >
      This is my fake title
    </h2>
    <p
      id="dialog-desc"
    >
      This is my fake description
    </p>
  </div>
</div>
`;

exports[`When user click on a button and modal should be displaying should render a Modal 1`] = `
<div
  aria-describedby="dialog-desc"
  aria-hidden="true"
  aria-labelledby="dialog-title"
  aria-modal="true"
  className="c-dialog"
  id="dialog"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="c-dialog__box"
    role="document"
  >
    <button
      aria-label="Close"
      className="btnClose"
      onClick={[Function]}
      title="Close window"
      type="button"
    >
      X
    </button>
    <h2
      id="dialog-title"
    >
      This is my fake title
    </h2>
    <p
      id="dialog-desc"
    >
      This is my fake description
    </p>
  </div>
</div>
`;

exports[`When user click on a button and modal shouldn't be displaying should not render a Modal 1`] = `
<div
  aria-describedby="dialog-desc"
  aria-hidden="false"
  aria-labelledby="dialog-title"
  aria-modal="true"
  className="c-dialog"
  id="dialog"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="c-dialog__box"
    role="document"
  >
    <button
      aria-label="Close"
      className="btnClose"
      onClick={[Function]}
      title="Close window"
      type="button"
    >
      X
    </button>
    <h2
      id="dialog-title"
    >
      This is my fake title
    </h2>
    <p
      id="dialog-desc"
    >
      This is my fake description
    </p>
  </div>
</div>
`;
